File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.*;
class Solution {
    public int solution(int[] arr) {
        int count=0;
        while(true){
            int[] preArr = new int[arr.length];

            for(int i=0; i<arr.length; i++){
                if(arr[i]%2==0 && arr[i]>=50){
                    preArr[i] = arr[i]/2;
                }
                else if(arr[i]%2!=0 && arr[i]<50){
                    preArr[i] = arr[i]*2+1;
                }
                else {preArr[i] = arr[i];}
            }
            
            if(Arrays.equals(arr,preArr)){
                return count;   
            }
            arr = preArr;
            count++;   

        }
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    count =0
    
    while True:
        preArr = arr[:]    
        arr = [i//2 if i >=50 and i%2==0 else(i*2+1 if i <50 and i%2!=0 else i) for i in arr]
        
        count += 1
        if arr == preArr:
            return count-1

